integration_tests/incentives_test.go

```diff
@@ -106,24 +138,66 @@ func (s *IntegrationTestSuite) TestIncentives() {
+		// Assumes each validator has equally weight bonding power
+		actualDistributionPerBlock := (afterAmount.Sub(beforeAmount)).Quo(sdk.NewDec(afterHeight - beforeHeight)).Mul(sdk.NewDec(int64(len(s.chain.validators))))
+		s.T().Logf("before: %s, after: %s, blocks %d-%d", beforeAmount.String(), afterAmount.String(), beforeHeight, afterHeight)
 		s.Require().True(afterAmount.GT(beforeAmount))
+		s.Require().Equal(expectedDistributionPerBlock.Amount.ToDec(), actualDistributionPerBlock)
+
+		s.T().Log("verifying APY query returns expected value")
+		mintParams, err := mintQueryClient.Params(ctx, &minttypes.QueryParamsRequest{})
+		s.Require().NoError(err)
+		stakingPool, err := stakingQueryClient.Pool(ctx, &stakingtypes.QueryPoolRequest{})
+		s.Require().NoError(err)
+		tokensTotal := stakingPool.Pool.BondedTokens.Add(stakingPool.Pool.NotBondedTokens)
+		// assumes bonded ratio is 100%
+		expectedAPY := actualDistributionPerBlock.Mul(sdk.NewDec(int64(mintParams.Params.BlocksPerYear))).Quo(tokensTotal.ToDec())
+		incentivesAPYRes, err = incentivesQueryClient.QueryAPY(ctx, &incentivestypes.QueryAPYRequest{})
+		s.Require().NoError(err)
+		APY, err := sdk.NewDecFromStr(incentivesAPYRes.Apy[:10])
+		s.Require().NoError(err)
+		// maxDiff, err := sdk.NewDecFromStr("0.00001")
+		// s.Require().NoError(err)
+		s.Require().Equal(expectedAPY, APY)
 	})
 }
 
-func (s *IntegrationTestSuite) queryDelegationRewardAmount(ctx context.Context, delegatorAddress string, distQueryClient disttypes.QueryClient) sdk.Dec {
-	rewardsRes, err := distQueryClient.DelegationRewards(ctx, &disttypes.QueryDelegationRewardsRequest{
-		DelegatorAddress: delegatorAddress,
-		ValidatorAddress: "sommvaloper199sjfhaw3hempwzljw0lgwsm9kk6r8e5ef3hmp",
+func (s *IntegrationTestSuite) queryValidatorRewards(ctx context.Context, valOperatorAddress string, distQueryClient disttypes.QueryClient) sdk.Dec {
+	rewardsRes, err := distQueryClient.ValidatorOutstandingRewards(ctx, &disttypes.QueryValidatorOutstandingRewardsRequest{
+		ValidatorAddress: valOperatorAddress,
 	})
 	s.Require().NoError(err)
-	return rewardsRes.Rewards.AmountOf(params.BaseCoinUnit)
+	return rewardsRes.Rewards.Rewards.AmountOf(params.BaseCoinUnit)
+}
+
+func (s *IntegrationTestSuite) getCurrentHeight(clientCtx *client.Context) int64 {
+	node, err := clientCtx.GetNode()
+	s.Require().NoError(err)
+	status, err := node.Status(context.Background())
+	s.Require().NoError(err)
+
+	return status.SyncInfo.LatestBlockHeight
+}
+
+func (s *IntegrationTestSuite) getRewardAmountAndHeight(ctx context.Context, distQueryClient disttypes.QueryClient, operatorAddress string, clientCtx *client.Context) (sdk.Dec, int64) {
+	var amount sdk.Dec
+	var height int64
+
+	s.Require().Eventually(func() bool {
+		initialHeight := s.getCurrentHeight(clientCtx)
+		amount = s.queryValidatorRewards(ctx, operatorAddress, distQueryClient)
+		currentHeight := s.getCurrentHeight(clientCtx)
+		height = currentHeight
```

Comment:
Nit: couldn't you just assign `height` directly without the intermediate variable here?